torch/csrc/jit/ir.h

```diff
@@ -933,7 +946,11 @@ friend struct Block;
     return current_scope_;
   }
   void set_current_scope(Scope* scope) {
-    if (scope->getRoot() != scope_root_.get()) {
+    std::string name1 = scope->getRoot()->name().toDisplayString();
+    std::string name2 = scope_root_.get()->name().toDisplayString();
+    if (name1.find(name2) < 0 &&
+        name2.find(name1) < 0 &&
+        scope->getRoot() != scope_root_.get()) {
```

Comment:
Those find will also find matches deeper down the scope chain. You probably want something that mimics the behavior of `startswith` better.